src/block/inventory/FurnaceInventory.php

```diff
@@ -68,4 +68,26 @@ public function setFuel(Item $item) : void{
 	public function setSmelting(Item $item) : void{
 		$this->setItem(self::SLOT_INPUT, $item);
 	}
+
+	public function canAddSmelting(Item $item) : bool{
+		if($this->getSmelting()->isNull()) return true;
+
+		$currentInput = $this->getSmelting();
```

Comment:
It's not that big of a deal, but you call `getSmelting()` twice here. Especially due to inventories cloning the item instance before returning it, this results in one unnecessary clone if the item is not null. So this should be changed to
```suggestion
		$currentInput = $this->getSmelting();
		if($currentInput->isNull()){
				return true;
		}
```